### PR TITLE
Swap order of access for FORMAT from `sample → field` to `field → sample`

### PR DESCRIPTION
**Pull request by tedil**

### CODE REVIEW

File: vembrane/__init__.py [hunk 1]
```diff
@@ -48,42 +52,55 @@
...
         except KeyError:
             try:
-                sample_format = self._sample[item]
+                record_sample = self._record_samples[sample]
             except KeyError as ke:
-                raise UnknownFormatField(self._record_idx, self._sample_name, ke)
-            self._data[item] = sample_format
-            return sample_format
+                raise UnknownSample(
+                    self._record_idx, str(ke).split(":")[1][1:].strip("'")
```
Comment (mbargull) [inline vembrane/__init__.py:72]: What does the `str(ke)` contain and what are you extracting? Something more than just `sample`?

Comment (tedil) [reply vembrane/__init__.py:72]: Yes, this is ugly, but the KeyError's sole arg is `'invalid sample name: SomeUnknownSampleName'`. Oh wow I could have just used `sample` here, nvm, will fix.

File: vembrane/__init__.py [hunk 2]
```diff
@@ -48,42 +52,51 @@
...
             try:
-                sample_format = self._sample[item]
-            except KeyError as ke:
-                raise UnknownFormatField(self._record_idx, self._sample_name, ke)
-            self._data[item] = sample_format
-            return sample_format
+                record_sample = self._record_samples[sample]
+            except KeyError:
+                raise UnknownSample(self._record_idx, sample)
+            value = record_sample[self._name]
```
Comment (mbargull) [inline vembrane/__init__.py:72]: `record.sample[fmt]` is guaranteed to exist, right?
In the old code we had `{fmt: record.samples[sample][fmt] for fmt in record.format}`, hence I assume it is guaranteed, but just checking/asking :).

Comment (tedil) [reply vembrane/__init__.py:72]: I'm not too sure to be honest. I always assumed all samples need to have every FORMAT field specified in the header (the VCF specification isn't too clear on this).
So at this point in the code, if `record.format[fmt]` didn't already KeyError, `record.samples[sample][fmt]` shouldn't either.

> LGTM (barring a missing test case :P).
We should definitely add more testcases.

**Pull request merged**